example/prometheus-operator-crd/monitoring.coreos.com_alertmanagerconfigs.yaml

```diff
@@ -0,0 +1,402 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.2.4
+  creationTimestamp: null
+  name: alertmanagerconfigs.monitoring.coreos.com
+spec:
+  group: monitoring.coreos.com
+  names:
+    kind: AlertmanagerConfig
+    listKind: AlertmanagerConfigList
+    plural: alertmanagerconfigs
+    singular: alertmanagerconfig
+  scope: Namespaced
+  versions:
+  - name: v1
```

Comment:
We don't want to promote this to v1 directly, lets instead change to v1beta1 version. cc @brancz @s-urbaniak we discussed this in a separate issue.

Reply:
@lilic Thanks for the feedback, I'm definitely up for changing it. :+1: 

Has there ever been a case previously in this operator where there was more than one version in the group? The reason I'm asking is that I'm having a little trouble figuring out how to get `client-gen` to generate `pkg/client/versioned/clientset.go` to contain both `v1` and `v1beta1`. It seems like it only accepts one version per group as input: https://github.com/kubernetes/code-generator/blob/master/cmd/client-gen/args/args.go#L78

Reply:
@s-urbaniak You may have suggestions here, as it's related to #2163

Reply:
Yes, I was thinking the same when I wrote the comment, honestly I don't think so. I believe because of the history all were promoted at once. @dgrisonnet touched some code generation in the past maybe he has more insight into it. Otherwise we can also build it twice maybe?

Reply:
@lilic I've trying running the `client-gen` command twice, with different `--input` values, if that's what you mean? Unfortunately that leads to just the version specified in the second invocation being used.

My thinking is that either we'd need to generate a different clientset for each version, something like `pkg/client/v1/versioned/` and `pkg/client/v1beta1/versioned/` side-by-side (I don't know what the knock-on effects of that are); or else do some post-processing (I don't know how much work that would be).

Another alternative might be to use a different group for v1beta1 (something like `monitoring/v1` but `prometheusoperator/v1beta1`? :thinking:

Reply:
@grdryn you should be able to do so using client-gen as it is the tool used to generate kubernetes' clientset.

I found an example in the apiserver-sample project. They are generating the [clientset](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/sample-apiserver/pkg/generated/clientset/versioned) for the wardle group v1alpha  and v1beta1 using the following [update-codegen.sh](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh) script. By digging into that script, you might be able to understand how to pass the input arguments correctly.

However, I think that we would be better off moving to code-generator's [generate-groups.sh](https://github.com/kubernetes/code-generator/blob/master/generate-groups.sh) script that is used in official kubernetes projects instead of maintaining our current custom generation scripts.

Reply:
Thanks @dgrisonnet that will definitely help. I'm starting to think that it actually works the way we need it to, even though it seems to say it doesn't. I'll hopefully get further with this today.

Reply:
Done (v1beta1).